src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+    // saved/restored during task switching
+
+    // DANGER, WILL ROBINSON!
+    ///
+    // This whole structure can be read or written from the guest side
+    // via virtual DOS driver, functions 0x15 / 0x16 / 0x17.
+    // Do not put here any array indices, pointers, or anything that
+    // can crash the emulator if filled-in incorrectly, or that can
+    // be used by malicious code to escape from emulation!
+
+    bool enabled    = false; // TODO: make use of this
+    bool cute_mouse = false;
+
+    uint16_t times_pressed[NUM_BUTTONS]   = {0};
+    uint16_t times_released[NUM_BUTTONS]  = {0};
+    uint16_t last_released_x[NUM_BUTTONS] = {0};
+    uint16_t last_released_y[NUM_BUTTONS] = {0};
+    uint16_t last_pressed_x[NUM_BUTTONS]  = {0};
+    uint16_t last_pressed_y[NUM_BUTTONS]  = {0};
+    uint16_t last_wheel_moved_x           = 0;
+    uint16_t last_wheel_moved_y           = 0;
+
+    float mickey_x = 0.0f;
+    float mickey_y = 0.0f;
+
+    float mickeys_per_px_x = 0.0f;
+    float mickeys_per_px_y = 0.0f;
+    float pxs_per_mickey_x = 0.0f;
+    float pxs_per_mickey_y = 0.0f;
+
+    uint16_t gran_x = 0; // granularity mask
+    uint16_t gran_y = 0;
+
+    int16_t update_region_x[2] = {0};
+    int16_t update_region_y[2] = {0};
+
+    uint16_t language = 0; // language for driver messages, unused
+    uint8_t mode      = 0;
+
+    // sensitivity
+    uint16_t senv_x_val       = 0;
+    uint16_t senv_y_val       = 0;
+    uint16_t dspeed_threshold = 0; // threshold, in mickeys/s  TODO: should
+                                   // affect mouse movement
+    float senv_x = 0.0f;
+    float senv_y = 0.0f;
+
+    // mouse position allowed range
+    int16_t minpos_x = 0;
+    int16_t maxpos_x = 0;
+    int16_t minpos_y = 0;
+    int16_t maxpos_y = 0;
+
+    // mouse cursor
+    uint8_t page                     = 0; // cursor display page number
+    bool inhibit_draw                = false;
+    uint16_t hidden                  = 0;
+    uint16_t oldhidden               = 0;
+    int16_t clipx                    = 0;
+    int16_t clipy                    = 0;
+    int16_t hot_x                    = 0; // cursor hot spot, horizontal
+    int16_t hot_y                    = 0; // cursor hot spot, vertical
+    bool background                  = false;
+    uint16_t backposx                = 0;
+    uint16_t backposy                = 0;
+    uint8_t backData[CURSOR_SIZE_XY] = {0};
+    MouseCursor cursor_type          = MouseCursor::Software;
+
+    // cursor shape definition
+    uint16_t text_and_mask                       = 0;
+    uint16_t text_xor_mask                       = 0;
+    bool user_screen_mask                        = false;
+    bool user_cursor_mask                        = false;
+    uint16_t user_def_screen_mask[CURSOR_SIZE_Y] = {0};
+    uint16_t user_def_cursor_mask[CURSOR_SIZE_Y] = {0};
+
+    // user callback
+    uint16_t sub_mask = 0;
+    uint16_t sub_seg  = 0;
+    uint16_t sub_ofs  = 0;
```

Comment:
What is `sub`?

Reply:
Subroutine address (segment, offset), and a bitmask - I have changed the names.